### PR TITLE
New Collection extension methods added

### PR DESCRIPTION
**Pull request by n0an**
I've added new methods to Collection extension:
* all() - Returns if all of collection elements conform to given condition
* any() - Returns if any of collection elements conforms to given condition
* none() - Returns if none of collection elements conforms to given condition
And added tests for these new methods.
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a changelog entry describing my changes.

### CODE REVIEW

**Comment by SD10** [comment]
@n0an a few things:
- First, you don't have to close your old PR whenever something goes wrong. You can just push your updated code to the branch you made your PR from and it will automatically update your PR. Just a tip ❤️ 

- Second, I'm pretty sure these already exist on `Array` because IIRC I was the one who wrote them. What I would do is move the implementations from `Array` to collection to remain consistent with the API's.

**Comment by n0an** [comment]
@SD10 , thank you
There went a lot of wrong with previous PR which I closed. My forked codebase was outdated and messed up.
Now I'll make clone right from native repo, thanks for advice.
Indeed, there're .all and .none methods in Array extension. I haven't noticed them before.

**Comment by SD10** [comment]
@n0an I was thinking about this today. Do you think that these methods can be delcared on `Sequence` instead? It's even more flexible than `Collection` then. This is related to #427

**Comment by n0an** [comment]
I think it's good idea really. Moved all(), any(), none() methods to newly created file SequenceExtensions.swift with Sequence Extension in it.
Maybe we should investigate what other methods can we move to Sequence from Array and Collection.

**Comment by SD10** [comment]
@n0an That's a good idea for sure. All that's missing from this is a CHANGELOG entry and I can merge

**Comment by n0an** [comment]
I've updated changelog.

**Comment by n0an** [comment]
@omaralbeik, I've resolved merge conflicts, as I understood them.. I'm not really sure if I've done this really correctly.  Please forgive if something wrong :)

File: Tests/SwiftStdlibTests/SequenceExtensionsTests.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+//  SwifterSwift
+//
+//  Created by Anton Novoselov on 04/04/2018.
+//  Copyright © 2018 SwifterSwift
+//
+
+import XCTest
+@testable import SwifterSwift
+
+class SequenceExtensionsTests: XCTestCase {
```
Comment (omaralbeik) [inline Tests/SwiftStdlibTests/SequenceExtensionsTests.swift]: Could you mark this as `final` please

Comment (n0an) [reply Tests/SwiftStdlibTests/SequenceExtensionsTests.swift]: Done this!

**Comment by omaralbeik** [comment]
Thank you for your quick response @n0an, looks good to me, let us just wait for Travis :)

**Pull request merged**